src/test/java/io/nats/client/impl/RequestTests.java

```diff
@@ -139,13 +141,41 @@ public void testMultipleRequest() throws IOException, ExecutionException, Timeou
         }
     }
 
+    @Test
+    public void testMultipleReplies() throws Exception {
+        Options.Builder builder = new Options.Builder().turnOnAdvancedStats();
+        runInServer(builder, nc -> {
+            AtomicInteger replies = new AtomicInteger();
+            MessageHandler handler = (msg) -> { replies.incrementAndGet(); nc.publish(msg.getReplyTo(), null); };
+            Dispatcher d1 = nc.createDispatcher(handler);
+            Dispatcher d2 = nc.createDispatcher(handler);
+            Dispatcher d3 = nc.createDispatcher(handler);
+            Dispatcher d4 = nc.createDispatcher(msg -> { sleep(5000); handler.onMessage(msg); });
```

Comment:
If you wanted to you could probably reduce the amount of time waiting in the test and wait on a latch in this dispatcher to make this more deterministic - could see this becoming a flapper on stressed systems.

Reply:
Yeah, it's not optimal, but also it's hard to latch onto a different thread. I think if it flaps, we should actually just give it more sleep time.